api/utils/marshmallow_schemas.py

```diff
@@ -0,0 +1,118 @@
+"""Validation schemas."""
+from datetime import date
+from marshmallow import (Schema, fields, post_load, validates,
+                         validate, ValidationError)
+from api.models import User, Activity, ActivityType
+
+
+class BaseSchema(Schema):
+    """Creates a base validation schema."""
+
+    uuid = fields.String(dump_only=True, validate=[
+        validate.Length(max=36)])
+    name = fields.String(
+        required=True,
+        error_messages={
+            'required': {'message': 'A name is required.'}
+        })
+    photo = fields.String()
+    created_at = fields.DateTime(dump_only=True)
+    modified_at = fields.DateTime(dump_only=True)
+    description = fields.String()
+
+
+class ActivityTypesSchema(BaseSchema):
+    """Creates a validation schema for activity types."""
+
+    description = fields.String(
+        required=True,
+        error_messages={
+            'required': 'A description is required.'
+        }
+    )
+    value = fields.Integer(
+        required=True,
+        error_messages={
+            'required': 'Please send the activity points value'
+        }
+    )
+
+
+class LoggedActivitySchema(BaseSchema):
+    """Creates a validation schema for logged activities."""
+
+    status = fields.String(dump_only=True)
+    value = fields.Integer(
+        required=True,
+        error_messages={
+            'required': 'Please send the activity points value'
+        }
+    )
+    user = fields.String(dump_only=True, attribute='user.name')
+    society = fields.String(dump_only=True, attribute='society.name')
+    approved_by = fields.Method('get_approver', dump_only=True)
+
+    @staticmethod
+    def get_approver(obj):
+        """Get approver details."""
+        if obj.approver_id:
+            approver = User.query.get(obj.approver_id)
+            return approver.name
+        return
+
+
+class ActivitySchema(BaseSchema):
+    """Creates a validation schema for activities."""
+
+    description = fields.String(
+        required=True,
+        error_messages={
+            'required': {'message': 'A description is required.'}
+        })
+    activity_type_id = fields.String(
```

Comment:
Might be also a good thing to validate the length of the id here. Since `uuids` are always 36 chars.

Reply:
@LarryWachira done.